kubernetes/structures.go

```diff
@@ -668,9 +668,7 @@ func flattenServiceAccountSecrets(in []api.ObjectReference, defaultSecretName st
 			continue
 		}
 		m := map[string]interface{}{}
-		if v.Name != "" {
```

Comment:
I don't think we need to update this one – just the `flattenLocalObjectReferenceArray` function above.